### DISCUSSION TITLE
Adjustible stage covers in osu!mania

### DISCUSSION THREAD

**Discussion by mcendu** [ppy/osu#15602] [Ideas]
Feature description
====
This feature works in a similar manner to the SUDDEN+ (cover the top portion) and HIDDEN+ (cover the bottom portion) mods from Beatmania IIDX, where a completely opaque drawable covers up a portion of the stage with this mod enabled. In the case of SUDDEN+, this reduces the time from the appearance of a hit object to when it reaches the judgement line, potentially making the beatmap easier to read similar to adjusting the speed.

This is similar to osu! Fade In and Hidden mods in functionality, but more customizable.

Proposed design
====
Implementation
----
I propose two different ways of implementing them: as a mod, or as a set of options.
- Pros of this being a mod:
  - Easy to make it submit as a part of a replay as the amount of area covered is implemented as a mod option
  - Could replace Fade In and Hidden (aesthetical difference is fine...?)
- Cons of this being a mod:
  - Configuration won't be saved. When the player switches game mode or exits the game it is lost
  - Hard to adjust in-game.
- Pros of this being part of the options:
  - Easy to adjust in-game - one just need to pause, Ctrl-O and navigate there to adjust the param
  - Configuration can be saved
- Cons of this being part of the options:
  - Might not be stored as part of the replay...?
  - Fade In and Hidden needs to be implemented separately, possibly referencing the lane cover functions, possibly not.

It is ok to implement one as a mod and the other as an option.

Regardless of implementation manner, I propose that the sliders for configuring the amount of screen hidden should have their range between 0 and 1 (both ends included), where 0 means no portion of the stage is hidden and 1 means that the stage is completely covered up. The value is ideally represented as a percentage, due to it being common and easy to understand.

Display
----
Due to it being a completely new feature, it is mostly definite that there would be no legacy skin support for this. Therefore we only need to consider the lazer skin as part of the design...?

Related to #3687.

**Comment by smoogipoo** [comment]
So if I understand correctly SUDDEN+ is alike Hidden and HIDDEN+ is alike Fade In?

I would like to see these as options rather than mods. Options will eventually be stored to replays.

**Comment by mcendu** [comment]
@arflyte Might I request input on this?
